apps/builder/app/builder/features/style-panel/sections/backgrounds/background-content.tsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * Will be fully rewritten in next iteration,
+ * as of now just implement feature parity with old backgrounds section
+ **/
+
+import type { StyleValue } from "@webstudio-is/css-data";
+import { theme, Flex, Grid, Label } from "@webstudio-is/design-system";
+import { ImageControl, TextControl } from "../../controls";
+import type { StyleInfo } from "../../shared/style-info";
+import type {
+  DeleteProperty,
+  SetProperty,
+  StyleUpdateOptions,
+} from "../../shared/use-style-data";
+
+import {
+  DeleteBackgroundProperty,
+  isBackgroundLayeredProperty,
+  isBackgroundStyleValue,
+  type SetBackgroundProperty,
+} from "./background-layers";
+
+import { FloatingPanelProvider } from "~/builder/shared/floating-panel";
+import { useRef } from "react";
+
+type BackgroundContentProps = {
+  currentStyle: StyleInfo;
+  setProperty: SetBackgroundProperty;
+  deleteProperty: DeleteBackgroundProperty;
+};
+
+const safeDeleteProperty = (deleteProperty: DeleteBackgroundProperty) => {
+  const result: DeleteProperty = (property, options) => {
+    // isBackgroundLayeredProperty is typeguard and ts don't understand === false
+    if (!isBackgroundLayeredProperty(property)) {
```

Comment:
```suggestion
    if (isBackgroundLayeredProperty(property) === false) {
```

Reply:
uhh didn't see the comment, wtf

Reply:
what if you assign to a constant first?

Reply:
Nothing except the above works

Reply:
here is how I would write it otherwise

```ts
const safeDeleteProperty = (deleteProperty: DeleteBackgroundProperty): DeleteProperty => {
  return (property, options) => {
    const isLayered = isBackgroundLayeredProperty(property);
    if (isLayered) {
      return deleteProperty(property, options);
    }
    throw new Error(`Property ${property} should be background style property`);
  };
};
```

Reply:
cc @Andarist another thing where typescript is definitely being crazy stupid, maybe you already know the issue for that?

Reply:
It's interesting that `=== false` doesn't work. I guess that it could, it's just probably `!` is more "idiomatic"/common in JS and that has been implemented in TS. 

The last suggestion by @kof works though (with both aliased `isLayered` and without that "alias"). So the "key" here is to flip the structure of the code and handle the error case later or to use `!`. 

Note that, in a similar vein, `=== true` doesn't work either (just in the flipped situation).

Reply:
Do u have a coding style that prescribed `=== false/true`?

Reply:
I think this is a terrible ts implementation that should be treated like a bug, it literally disrespects javascript's language semantics

Reply:
how do I create a minimal reproducible case for this?

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAlhAgsKcBuBTGBeGAzAQwBsIMAoUSWNYgVy1wAoBKHAPhgG8yZ48ZGCZKkytuvXgCcMUWpLA8YAXxiLps+VygBPAA4YAXDACMSsmbJx+gpCnQNsuQiQxjFNIvRYA6HfoDc5mSW1gCEQnaiXO50GD5+GIFKQA

Reply:
Minimal repro for the said issue:
https://www.typescriptlang.org/play?ts=5.1.0-dev.20230308#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXywGcAVGZDACwE8AKHAIwCsAueNAa1RwHdUBKVg0YFC8AN4BfAFBTQkWAjB5CGeENZikOHKxUwsqAOYBueBPgAfeKmQQIMrIng0ipctTpM+fcVPhqmADpEbSlpKUdnVzJKWiEfAF4k+AwyEB8xPwDGYNCJIA